docs/middleware/BLSPublicKeyCompendium.md

```diff
@@ -0,0 +1,23 @@
+# BLSPublicKeyCompendium
+
+This contract is shared by all AVSs and serves a single place for operators to connect their execution layer address to a bn254 public key. When operators opt into AVSs, the AVS contracts can read from the BLSPublicKeyCompendium in order to get the BLS public key of the operator. This contract also prevents against [rogue key attacks](https://xn--2-umb.com/22/bls-signatures/#rogue-key-attack).
+
+## Flows
+
+There is only one flow for this contract, which is a call from an operator to register their execution layer address with thier bn254 public key.
+
+### Cryptographic Verification
+
+Operators provide the contract with 
+- A BLS signature $\sigma \in \mathbb{G}_1$ of $M = (msg.sender, chain.id, \text{"EigenLayer\_BN254\_Pubkey\_Registration"})$
+- Their public keys $pk_1 \in \mathbb{G}_1$ and $pk_2 \in \mathbb{G}_2$
+
+The contract then 
+- Calculates $\gamma = keccak256(\sigma, pk_1, pk_2, M)$
+- Verifies $e(\sigma + \gamma pk_1, [1]_2) = e(H(m) + \gamma[1]_1, pk_2)$
```

Comment:
verifies _the pairing_....
would also help to explain conceptually what this is doing rather than just writing an equation

Reply:
Here: d88f9e53

What do you want explanation wise?

Reply:
ehh I guess it's explained below, will resolve this thread